modules/geo/src/internalClusterTest/java/org/opensearch/geo/search/aggregations/bucket/AbstractBucketAggregationIntegTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.search.aggregations.bucket;
+
+import org.opensearch.Version;
+import org.opensearch.action.index.IndexRequestBuilder;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.geo.GeoModulePluginIntegTestCase;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.test.VersionUtils;
+
+import java.util.List;
+
+import static org.opensearch.common.xcontent.XContentFactory.jsonBuilder;
+
+/**
+ * This is the base class for all the Bucket Aggregation related integration tests. Use this class to add common
+ * methods which can be used across different bucket aggregations. If there is any common code that can be used
+ * across other integration test too then this is not the class. Use {@link GeoModulePluginIntegTestCase}
+ */
+public abstract class AbstractBucketAggregationIntegTest extends GeoModulePluginIntegTestCase {
+
+    protected static final int MAX_PRECISION_FOR_GEO_SHAPES_AGG_TESTING = 4;
+
+    protected static final int NUM_DOCS = 100;
+
+    protected static final String GEO_SHAPE_INDEX_NAME = "geoshape_index";
+
+    // We don't want to generate this BB at random as it may lead to generation of very large or very small BB.
```

Comment:
It's true we don't want the test to be too demanding but we also don't want to miss potential edge cases caught by randomization. Maybe add a new method `GeoTestUtil.randomBox(double radius)`?

```java
private static Rectangle randomRectangle(double radius) {
    double centerLat = GeoTestUtil.nextLatitude();
    double centerLon = GeoTestUtil.nextLongitude();
    Polygon p = GeoTestUtil.createRegularPolygon(centerLat, centerLon, radius, 4);
    return new Rectangle(p.minLat, p.maxLat, p.minLon, p.maxLon);
}
```

If I remember right, this test util may generate irregular line polygons, so you may have to take some inspiration from lucene's `GeoTestUtil#nextBoxInternal(boolean canCrossDateline)`

Reply:
@nknize I think generating a BBOX with a specified width is a good idea.

Reply:
sure will do that. I used a little different logic to prepare the BBox with a specific radius.